src/python/dicomifier/dicom_to_nifti/stacks.py

```diff
@@ -415,14 +415,14 @@ def canon_getter(data_set, tag):
             br"\)$",
             image_comments)
         if not match:
-            logger.debug("ImageComments not matched: '{}'".format(image_comments))
+            logger.debug("ImageComments not matched: '%s'" % image_comments)
             return None
         
         try:
-            b_value_comment, x, y, z = [float(x) for x in match.groups()]
+            b_value_comment, x, y, z = (float(x) for x in match.groups())
         except ValueError:
             logger.debug(
-                "b-value discrepancy: {} != {}".format(
+                "b-value discrepancy: %s != %f" % (
```

Comment:
I hope `%s` and `%f` are correct here.

Reply:
`b_value_element` should be a float and `b_value_comment` a string. However, this codes deals with a hack used by a vendor to store diffusion-related information: it may be wiser to keep the eager string interpolation, since this will work whatever the actual type of its arguments.

Reply:
Changed it to an f-string.